src/requestor/routes/query.py

```diff
@@ -106,10 +106,14 @@ async def list_requests(
 
     # get the resources the current user has access to see
     token_claims = await auth.get_token_claims()
-    # TODO update this endpoint to accept client tokens. We need to get the
-    # auth mapping for the client instead of the user
-    username = token_claims["context"]["user"]["name"]
-    authz_mapping = await api_request.app.arborist_client.auth_mapping(username)
+    username = token_claims.get("context", {}).get("user", {}).get("name")
+    if username:
+        authz_mapping = await api_request.app.arborist_client.auth_mapping(username)
+    else:
+        client_id = token_claims.get("azp")
+        authz_mapping = await api_request.app.arborist_client.client_auth_mapping(
```

Comment:
Arborist returns a 400 error "must specify exactly one of `username` or `clientID`"

Reply:
Can you update the openapi docs with that as a possible error response?

Reply:
The responses in the openapi docs are generated automatically and i'm not sure how to add a new one 🤔 But i could update the code to catch this and return a 401 directly, instead of passing None to arborist

Reply:
hmm, yeah if `None` will never make sense to even send, let's catch that here instead of waiting for the error downstream

Reply:
done